com.unity.netcode.gameobjects/Components/NetworkAnimator.cs

```diff
@@ -832,7 +832,8 @@ private void CheckForStateChange(int layer)
                     stateChangeDetected = true;
                     //Debug.Log($"[Cross-Fade] To-Hash: {nt.fullPathHash} | TI-Duration: ({tt.duration}) | TI-Norm: ({tt.normalizedTime}) | From-Hash: ({m_AnimationHash[layer]}) | SI-FPHash: ({st.fullPathHash}) | SI-Norm: ({st.normalizedTime})");
                 }
-                else if (!tt.anyState && tt.fullPathHash != m_TransitionHash[layer])
+                else if (!tt.anyState && tt.fullPathHash != m_TransitionHash[layer] && (!m_DestinationStateToTransitioninfo.ContainsKey(layer) ||
+                    (m_DestinationStateToTransitioninfo.ContainsKey(layer) && m_DestinationStateToTransitioninfo[layer].ContainsKey(nt.fullPathHash))))
```

Comment:
Nitpick: I wonder if we can split the condition into simpler checks, which makes it easier to understand.

```
else if (!tt.anyState && tt.fullPathHash != m_TransitionHash[layer])
{
  bool containsLayer = m_DestinationStateToTransitioninfo.ContainsKey(layer);
  bool containsFullPathHash = containsLayer && m_DestinationStateToTransitioninfo[layer].ContainsKey(nt.fullPathHash);

  if (!containsLayer ||  containsFullPathHash)
    {
       // first time in this transition for this layer
       m_TransitionHash[layer] = tt.fullPathHash;
    }
```

Reply:
Yeah... it would really be nice to have access to some of the Animator stuff that is available in the editor at runtime...but it gets boiled down to these condensed states.  

Although, I think you made me think more about this and that it probably needs some comments... 
👍

Reply:
Added some additional comments about that crazy logic... but basically it is avoiding from sending a animation transition that was a trigger to state or trigger from state back to Idle(state) transition which is already sent when the trigger was initially set.